### ISSUE TITLE
Is it possible to get the progress of uploaded data?

### ISSUE THREAD

**Issue by Dalinar651** [rikulo/socket.io-client-dart#261]
I am uploading a big file to aws s3 storage using sockets. In the meantime i want the user to show a progress bar of the uploaded data .

> This is the code i am using.
`var file=File().openRead();
file.listen((bytes){
socket.emitWithAck('upload-to-s3',bytes,ack:(data){},binary:true);
});`

**Comment by jumperchen** [comment]
Yes, you can use `ack` to simulate the progress when the callback of the `ack` is invoked.
